### PR TITLE
Adding cosmic_profiles

### PR DESCRIPTION
**Pull request by tibordome**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by tibordome** [comment]
I have been struggling with the openmp part of this conda build for quite some time now. I have checked the knowledge base documentation and various forums. Even though I have added libgomp to my build and host dependency list, I keep getting a “but ['_openmp_mutex'] not in reqs/run” error. Also, locally, “conda build .” passes, and “but ['_openmp_mutex'] not in reqs/run” basically becomes a warning. Could you please help me with this, 

@conda-forge-admin, please ping team

**Comment by tibordome** [comment]
@conda-forge-admin, please ping team

**Comment by tibordome** [comment]
@conda-forge-admin, please ping team

thank you @ngam for providing suggestions and pointing me to grayskull

Locally build succeeds, but here the license file cannot be found for some reason. This might be since in the pypi installation v.1.2.0, https://pypi.org/project/cosmic-profiles/1.2.0/, I have not added "license_files = ('LICENSE.md',)," to setup.py, which I only found later. Do you think a new PyPi installation is necessary, v.1.2.1, which has the aforementioned addition to setup.py?

**Comment by tibordome** [comment]
@ngam, i think i might need some help indeed, the tests are not very extensive so I should probably get them working.

the docker builds seem very sensitive, the last one was bickering about "Unable to start a daemon on the local node", could you help with that?

**Comment by tibordome** [comment]
@ngam I fixed the testing part. Finally.

Afterwards I get, 
ompi_mpi_init: ompi_rte_init failed
  --> Returned "Unable to start a daemon on the local node" (-127) instead of "Success" (0)
Which I have seen on computing clusters before, but not sure how to resolve it..

**Comment by tibordome** [comment]
@ngam, the error I mentioned shows up even if I do not run pytest, due to openmpi as I have found. Most likely this is a bug described here, https://github.com/open-mpi/ompi/issues/7330. I tried using version v.4.0.1 of openmpi with no success. How can i install packages like ssh or rsh and how can i export environment variables on the host?

File: recipes/cosmic_profiles/meta.yaml [hunk 1]
```diff
@@ -52,9 +53,8 @@ test:
     - cosmic_profiles/tests/test_shapes_ex_script.py
   requires:
     - pytest
-    - pip
   commands:
-    - pip check
```
Comment (ocefpaf) [inline recipes/cosmic_profiles/meta.yaml]: Is `pip check` failing? We recommend to keep it, along with the other tests, unless there is some known error.

File: recipes/cosmic_profiles/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+test:
+  imports:
+    - cosmic_profiles
+  source_files:
+    - cosmic_profiles/tests
+  requires:
+    - pytest
+    - pip
+  commands:
+    - pip check
```
Comment (ocefpaf) [inline recipes/cosmic_profiles/meta.yaml:57]: You should keep the full tests too: `pytest cosmic_profiles/tests`

**Comment by tibordome** [comment]
@ngam, @ocefpaf i haven't yet raised the issue on gitter, no.

i think i resolved the plm_rsh_agent-related error by choosing v.4.0.4 of openmpi, but then I got some hdf5-related error, "undefined symbol: H5Pset_fapl_ros3".

**Comment by tibordome** [comment]
@ngam https://github.com/h5py/h5py/issues/1880 was very useful thank you.

linux_64 ran successfully now.

File: recipes/cosmic_profiles/meta.yaml [hunk 3]
```diff
@@ -42,6 +42,7 @@ requirements:
     - h5py <3.2
     - psutil
     - openmpi =4.0.4  # [linux]
+    - mpich  # [osx]
```
[the remark this answers is not in the document]
Comment (tibordome) [reply recipes/cosmic_profiles/meta.yaml:45]: I resolved the libmpi.12.dylib problem by installing mpich. I did not try openmpi =4.0.4 for OSX, but the error does not appear now.

**Comment by tibordome** [comment]
@ngam, at the moment it is just the Windows testing that is failing. I am not a Windows expert, but it looks like some Python lines such as "subprocess.call(['mkdir', 'viz'], cwd=os.path.join(currentdir))" would fail in a Windows terminal. 

I have commented windows out from the pytest line if that's ok.

File: recipes/cosmic_profiles/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  imports:
+    - cosmic_profiles
+  source_files:
+    - cosmic_profiles/tests
+  requires:
+    - pytest
+    - pip
+  commands:
+    - pip check
+    - pytest cosmic_profiles/tests  # [linux and osx]
```
Comment (ocefpaf) [inline recipes/cosmic_profiles/meta.yaml:59]: Can we skip the tests that are expected to fail on Windows? You can do that in the feedstock. You can use `-k "not ( name-of-the-test)"`

Comment (tibordome) [reply recipes/cosmic_profiles/meta.yaml:59]: @ocefpaf good to know. Many thanks for merging!

**Pull request merged**